packages/concerto-core/lib/introspect/validator.js

```diff
@@ -46,10 +48,11 @@ class Validator {
     /**
      * @param {string} id the identifier of the instance
      * @param {string} msg the exception message
+     * @param {string} errorType the type of error
      * @throws {Error} throws an error to report the message
      */
-    reportError(id, msg) {
-        throw new Error( 'Validator error for field `' + id + '`. ' + this.getFieldOrScalarDeclaration().getFullyQualifiedName() + ': ' + msg );
+    reportError(id, msg, errorType='ValidatorError') {
```

Comment:
Let's avoid having to use `instanceof` checks - those make assumptions about how packages are required (instanceof can fail if the same code is loaded at different version) which have bitten us in the past.

Reply:
Hi @ekarademir I have used the additional argument `errorType` so that it can be propagated through till `StringValidationException` where we can use it to denote what kind of error it is , which can be very useful to do any kind of operation as only error string is quick tricky to detect what error it is trying to throw

Reply:
Hi @dselman , can you please clarify what do you mean by the instance of checks?
Thanks.

Reply:
I mean client code doing things like:

```
try {
...
}
catch(err) {
if(err instanceof StringValidatorException) {
 ...
}
}
```